src/FlatbuffersTranslator.cpp

```diff
@@ -0,0 +1,75 @@
+#include "FlatbuffersTranslator.h"
+#include "ArgumentsException.h"
+#include <boost/filesystem.hpp>
+
+#include <iostream>
+
+void FlatbuffersTranslator::getFileID(std::string *Message) {
+  // get the ID from a message
+  std::string FileID = Message->substr(4, 4);
+
+  if (FileIDMap.find(FileID) ==
+      FileIDMap.end()) { // if no ID present in the map:
+
+    // get schema name and path for FileID
+
+    std::string SchemaFile = getSchemaPathForID(FileID);
+    std::string Schema;
+    bool ok = flatbuffers::LoadFile(SchemaFile.c_str(), false, &Schema);
+    if (!ok) {
+      // Logger->error("Couldn't load schema files!\n");
```

Comment:
Looks like a good idea to log this. Note, you can get the logger you created in `main()` like this: https://github.com/gabime/spdlog/wiki/2.-Creating-loggers#accessing-loggers-using-spdlogget